### PR TITLE
(docs/statsd) updated doc with new configuration detail

### PR DESCRIPTION
**Pull request by shashiranjan84**

### CODE REVIEW

**Comment by shashiranjan84** [comment]
Merge only after https://github.com/Mashape/kong/pull/2090 get merged.

File: app/plugins/statsd.md [hunk 1]
```diff
@@ -42,8 +41,7 @@ parameter                     | default | description
 `consumer_id`<br>*optional*   |         | The CONSUMER ID that this plugin configuration will target. This value can only be used if [authentication has been enabled][faq-authentication] so that the system can identify the user making the request.
 `config.host`<br>*optional*   | `127.0.0.1` | The IP address or host name to send data to
 `config.port`<br>*optional*   | `8125`  | The port to send data to on the upstream server
-`config.metrics`<br>*optional* | All metrics<br>are logged | The metrics to be logged. Available values are described at [Metrics](#metrics).
-`config.timeout`<br>*optional* | `10000` | An optional timeout in milliseconds when sending data to the upstream server
+`config.metrics`<br>*optional* | All metrics<br>are logged | List of Metrics to be logged. Available values are described at [Metrics](#metrics).
```
Comment (nijikokun) [inline app/plugins/statsd.md]: >described at

should be

>described under

File: app/plugins/statsd.md [hunk 2]
```diff
@@ -54,17 +52,40 @@ parameter                     | default | description
 
 ## Metrics
 
-Plugin currently logs four metrics to the StatsD server: `request_count`, `request_size`, `response_size`, `status_count` and `latency`.
+Plugin currently logs following metrics to the Statsd server.
```
Comment (nijikokun) [inline app/plugins/statsd.md]: Should be:

> Metrics the plugin supports logging to the StatsD server.

File: app/plugins/statsd.md [hunk 3]
```diff
@@ -54,17 +52,40 @@ parameter                     | default | description
...
+Field         | description                                                | allowed values
+---           | ---                                                        | --- 
+`name`          | StatsD metric's name                                     | [Metrics](#metrics)          
+`stat_type`     | determines what sort of event the metric represents      | `gauge`, `timer`, `counter`, `histogram`, `meter` and `set`|
+`sample_rate`<br>*conditional*   | sampling rate                           | `number`                 
+`customer_identifier`<br>*conditional*| authenticated user detail          | `consumer_id`, `custom_id`, `username`
+
+### Metric Requirements
+
+1.  by default all metrics get logged.
```
Comment (nijikokun) [inline app/plugins/statsd.md]: `by` should be `By`

File: app/plugins/statsd.md [hunk 4]
```diff
@@ -54,17 +52,40 @@ parameter                     | default | description
...
+2.  metric with `stat_type` as `counter` or `gouge` must have `sample_rate` defined as well.
```
Comment (nijikokun) [inline app/plugins/statsd.md]: Should read like:

> Metrics with `stat_type` set to ...

**Comment by shashiranjan84** [comment]
@nijikokun thanks, just updated the file as advised. Please review.

**Pull request closed without merging**